datadog/provider.go

```diff
@@ -97,17 +114,20 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	communityClient.ExtraHeader["User-Agent"] = fmt.Sprintf("terraform-provider-datadog/%s (go %s; terraform %s; terraform-cli %s)", version.ProviderVersion, runtime.Version(), meta.SDKVersionString(), datadogProvider.TerraformVersion)
 	communityClient.HttpClient = c
 
-	log.Println("[INFO] Datadog client successfully initialized, now validating...")
-	ok, err := communityClient.Validate()
-	if err != nil {
-		log.Printf("[ERROR] Datadog Client validation error: %v", err)
-		return nil, err
-	} else if !ok {
-		err := errors.New(`Invalid or missing credentials provided to the Datadog Provider. Please confirm your API and APP keys are valid and are for the correct region, see https://www.terraform.io/docs/providers/datadog/ for more information on providing credentials for the Datadog Provider`)
-		log.Printf("[ERROR] Datadog Client validation error: %v", err)
-		return nil, err
+	if validate {
+		log.Println("[INFO] Datadog client successfully initialized, now validating...")
+		ok, err := communityClient.Validate()
+		if err != nil {
+			log.Printf("[ERROR] Datadog Client validation error: %v", err)
+			return nil, err
+		} else if !ok {
+			err := errors.New(`Invalid or missing credentials provided to the Datadog Provider. Please confirm your API and APP keys are valid and are for the correct region, see https://www.terraform.io/docs/providers/datadog/ for more information on providing credentials for the Datadog Provider`)
+			log.Printf("[ERROR] Datadog Client validation error: %v", err)
+			return nil, err
+		}
+	} else {
+		log.Println("[INFO] Skipping key validation (validate = false)")
 	}
-	log.Printf("[INFO] Datadog Client successfully validated.")
 
 	// Initialize the official Datadog V1 API client
 	authV1 := context.WithValue(
```

Comment:
We should use the new apiKey and appKey variables for the other client initialization as well. (This is a bit outside the scope of your changes, so we can always do this later :))